### PR TITLE
PR 1 - added misc changes requested by UI Team in contracts

### PR DESCRIPTION
**Pull request by shivamgarglab49**

### CODE REVIEW

File: e2e-test/business/MultiSigDao.ts [hunk 1]
```diff
@@ -112,6 +117,25 @@ export default class MultiSigDao extends BaseDao {
...
+    client: Client = clientsInfo.operatorClient
+  ) => {
+    const args = new ContractFunctionParameters().addBytes32(txnHash);
+    const { result } = await this.execute(
+      5_00_000,
+      GET_APPROVAL_COUNTS,
+      client,
+      args
+    );
+    const hash = ethers.utils.hexlify(txnHash);
```
Comment (amandeeprehal-lab49) [inline e2e-test/business/MultiSigDao.ts:131]: Do you need hexlify here or its required for Boolean?

Comment (shivamgarglab49) [reply e2e-test/business/MultiSigDao.ts:131]: this is just for debugging purpose, we using it in every function call like getTransactionInfo, state etc.

File: deployment/state/contracts.json [hunk 2]
```diff
@@ -32,45 +24,45 @@
     "hash": "011a69486f8974e7811380eecddbc3bd",
     "transparentProxyAddress": "0x0000000000000000000000000000000000d16e56",
     "transparentProxyId": "0.0.13725270"
+=======
```
[the remark this answers is not in the document]
Comment (shivamgarglab49) [reply deployment/state/contracts.json]: fixed

File: contracts/governance/GODHolder.sol [hunk 3]
```diff
@@ -43,6 +43,7 @@ contract GODHolder is TokenHolder {
             code == HederaResponseCodes.SUCCESS,
             "GODHolder: token transfer failed from contract."
         );
+        emit UpdatedAmount(msg.sender, godTokenForUsers[msg.sender], 1);
```
[the remark this answers is not in the document]
Comment (shivamgarglab49) [reply contracts/governance/GODHolder.sol]: done

File: utils/InstanceProvider.ts [hunk 4]
```diff
@@ -110,9 +115,22 @@ export class InstanceProvider {
...
 
-  public getFungibleTokenHolder(id: string | null = null) {
-    const _id = this.getProxyId(id, this.csDev.godHolderContract);
-    return new GodHolder(_id);
+  public async getNFTTokenHolderFromFactory(
+    tokenId: TokenId,
+    id: string | null = null
+  ) {
+    const factory = this.getNFTTokenHolderFactory(id);
+    const cId = await factory.getTokenHolder(tokenId.toSolidityAddress());
```
Comment (amandeeprehal-lab49) [inline utils/InstanceProvider.ts:117]: This dependency on business function inside instance provider does not make sense.

Comment (shivamgarglab49) [reply utils/InstanceProvider.ts:117]: sure

**Review by amandeeprehal-lab49 — APPROVED** [review]

**Pull request merged**